File: 백준/Gold/5430. AC/AC.py
import sys
from collections import deque

T = int(input())
isError = False

for _ in range(T):
    p = sys.stdin.readline().rstrip()
    n = int(sys.stdin.readline().rstrip())
    nums = deque(sys.stdin.readline().rstrip().replace('[', '').replace(']', '').replace(',', ' ').split())

    count = 0
    for i in range(len(p)):
        if p[i] == 'R':
            count += 1
        else:
            if count % 2 == 0:
                if nums:
                    nums.popleft()
                else:
                    sys.stdout.write("error" + "\n")
                    isError = True
                    break
            else:
                if nums:
                    nums.pop()
                else:
                    sys.stdout.write("error" + "\n")
                    isError = True
                    break

    if not isError:
        if len(nums) == 0:
            sys.stdout.write("[]" + "\n")
        else:
            if count % 2 == 0:
                sys.stdout.write("[")
                for i in range(len(nums) - 1):
                    sys.stdout.write(str(nums[i]) + ",")
                sys.stdout.write(str(nums[-1]) + "]" + "\n")
            else:
                sys.stdout.write("[")
                for i in range(len(nums) - 1, 0, -1):
                    sys.stdout.write(str(nums[i]) + ",")
                sys.stdout.write(str(nums[0]) + "]" + "\n")

    nums.clear()
    isError = False
    count = 0

File: 백준/Gold/5430. AC/AC.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringBuilder sb = new StringBuilder();

        int t = Integer.parseInt(br.readLine());

        for (int i = 0; i < t; i++) {
            char direction = 'L';
            String[] p = br.readLine().split("");
            int n = Integer.parseInt(br.readLine());
            Deque<Integer> q = new ArrayDeque<>();
            String inputLine = br.readLine();
            if (n > 0) {
                String[] inputData = inputLine.replace("[", "").replace("]", "").split(",");
                for (int j = 0; j < n; j++) {
                    q.add(Integer.parseInt(inputData[j].trim()));
                }
            }

            boolean error = false;
            for (String command : p) {
                switch (command) {
                    case "R":
                        direction = (direction == 'L') ? 'R' : 'L';
                        break;
                    case "D":
                        if (q.isEmpty()) {
                            error = true;
                            break;
                        }
                        if (direction == 'L') {
                            q.pollFirst();
                        } else {
                            q.pollLast();
                        }
                        break;
                }
                if (error) {
                    break;
                }
            }

            if (error) {
                sb.append("error\n");
            } else {
                sb.append("[");
                if (!q.isEmpty()) {
                    if (direction == 'L') {
                        while (q.size() > 1) {
                            sb.append(q.pollFirst() + ",");
                        }
                        sb.append(q.pollFirst());
                    } else {
                        while (q.size() > 1) {
                            sb.append(q.pollLast() + ",");
                        }
                        sb.append(q.pollLast());
                    }
                }
                sb.append("]\n");
            }
        }
        bw.write(sb.toString());
        bw.close();
        br.close();
    }
}
